WireAPI/Sources/WireAPI/Private/Extensions/Date+Transport.swift

```diff
@@ -0,0 +1,41 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension Date {
+
+    /// Initialze the date from the given transport string.
+
+    init?(transportString: String) {
+        if let date = ISO8601DateFormatter.default.date(from: transportString) {
+            self = date
+        } else if let date = ISO8601DateFormatter.withoutFractionalSeconds.date(from: transportString) {
+            self = date
+        } else {
+            return nil
+        }
+    }
```

Comment:
This duplicates the date decoding code, that has had bugs in the last time, here in the new swift package. Hm.

Wouldn't be the right approach, that we know for all APIs, that we move here to the package, what kind of `Date` format we expect by writing unit tests?

My goal would be that we get rid of this generalized helper here long term and preferably use `jsonDecoder.dateDecodingStrategy = .iso8601`. In rare cases where the backend delivers fractional seconds and it can not be changed, we can use `.custom`  with the `ISO8601DateFormatter` as workaround.

Reply:
I think the backend always delivers fractional seconds, which is why the default json decoder uses the custom date decoding strategy.

Reply:
Probably I mixed something up here, the implementation in `WireTransport` has also a slightly different naming.
The cleanup etc. is also something for a follow up step, so I resolve the conversation here.